scripts/perf_target_scale.py

```
@@ -0,0 +1,809 @@
+        while pending and time.monotonic() < deadline:
+            for sid in list(pending):
+                try:
+                    text = _read_surface_text(self.rpc("surface.read_text", {"workspace_id": workspace_id, "surface_id": sid}, timeout=30))
+                except BenchmarkError:
+                    continue
+                if "CMUX_TARGET_SCALE_READY" in text:
+                    pending.remove(sid)
+            if pending:
+                time.sleep(0.2)
+        if pending:
+            raise BenchmarkError(f"scrollback seed did not settle for {len(pending)} surfaces")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)